### PR TITLE
Improve Local file system and the fileSystem interface 

### PR DESCRIPTION
**Pull request by alisihab**

### CODE REVIEW

**Comment by gvanbrakel** [comment]
- Please remove SambaFileSystem.java from this PR, it is out of scope and not yet fully functional
- Please add LocalFileSystemSender and it's test methods.

LocalFileSystem.java
- renameTo() should throw an exception when target file exists and force=false
- please remove //TODO Auto-generated method stub comments in open() and close(), replace with comment that no action is required

FileSystemTest.java
- please add in all the comments of the methods with underscore
- waitForActionToFinish(): please do something with the InterruptedException. Never catch an Exception and ignoring, without even a comment why it can be ignored!
- testNotExist() does not need to create the file before deleting it.
- avoid double //   //   comments
- What is the meaning of testExistsContinue()? It appears to be the same as testExists(). If useful, please document, otherwise remove
- existsCheck(): do not need to catch exception when it is only rethrown
- testAppendFile(), and also other test methods: please separate the parts for 'setup', 'execute' and 'assert' by whitespace and a comment line.
- setWaitMilis(): please add extra 'l': millis in stead of milis

File: core/src/main/java/nl/nn/adapterframework/filesystem/LocalFileSystem.java [hunk 1]
```diff
@@ -109,9 +106,70 @@ public void removeFolder(File f) {
...
+	}
+
+	@Override
+	public Date getModificationTime(File f, boolean isFolder) throws FileSystemException {
+		return new Date(f.lastModified());
+	}
+
+	@Override
+	public void augmentDirectoryInfo(XmlBuilder dirInfo, File f) {
+		dirInfo.addAttribute("name", f.getName());
```
[the remark this answers is not in the document]
Comment (alisihab) [reply core/src/main/java/nl/nn/adapterframework/filesystem/LocalFileSystem.java]: Currently, we use augmentDirectoryInfo method in list action. Since we don't have a test class for LocalFileSystemSender, we can leave this method empty. However, when we create test class for the class it will fail listActionTest. It needs to have the same structure as other filesystem classes have because in list action test we need to be able parse count in the same way we do now.
